config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/JmsDestinationEntityBuilder.java

```diff
@@ -0,0 +1,281 @@
+            }
+
+            jmsDestinationDetailProps.put(USE_REQUEST_CORRELATION_ID, inboundDetail.useRequestCorrelationId());
+
+            ServiceResolutionSettings serviceResolutionSettings = inboundDetail.getServiceResolutionSettings();
+            if (serviceResolutionSettings != null) {
+                String serviceRef = serviceResolutionSettings.getServiceRef();
+                if (serviceRef != null) {
+                    Service service = bundle.getServices().get(serviceRef);
+                    if (service == null) {
+                        throw new EntityBuilderException("Could not find associated Service for inbound JMS Destination: " + name + ". Service Path: " + serviceRef);
+
+                    }
+                    contextPropertiesTemplateProps.put(IS_HARDWIRED_SERVICE, true);
+                    contextPropertiesTemplateProps.put(HARDWIRED_SERVICE_ID, service.getId());
+                } else {
+                    contextPropertiesTemplateProps.put(IS_HARDWIRED_SERVICE, false);
+                }
+
+                putToMapIfValueIsNotNull(
+                        contextPropertiesTemplateProps,
+                        SOAP_ACTION_MSG_PROP_NAME,
+                        serviceResolutionSettings.getSoapActionMessagePropertyName());
+
+                String contentTypeSource;
+                switch (serviceResolutionSettings.getContentTypeSource()) {
+                    case NONE:
+                        contentTypeSource = "";
+                        break;
+                    case FREE_FORM:
+                        contentTypeSource = "com.l7tech.server.jms.prop.contentType.freeform";
+                        break;
+                    case JMS_PROPERTY:
+                        contentTypeSource = "com.l7tech.server.jms.prop.contentType.header";
+                        break;
+                    default:
+                        contentTypeSource = "";
+                        break;
+                }
+                contextPropertiesTemplateProps.put(CONTENT_TYPE_SOURCE, contentTypeSource);
+                
+                String contentType = serviceResolutionSettings.getContentType();
+                if (contentType == null) {
+                    contentType = "";
+                }
+                contextPropertiesTemplateProps.put(CONTENT_TYPE_VALUE, contentType);
+            }
+            
+            if (ON_COMPLETION.equals(inboundDetail.getAcknowledgeType())){
+                putToMapIfValueIsNotNull(
+                        jmsDestinationDetailProps,
+                        INBOUND_FAILURE_QUEUE_NAME,
+                        inboundDetail.getFailureQueueName());
+            }
+
+            contextPropertiesTemplateProps.put(IS_DEDICATED_CONSUMER_CONNECTION, true);
+            Integer numOfConsumerConnections = inboundDetail.getNumOfConsumerConnections();
+            if (numOfConsumerConnections == null) {
+                numOfConsumerConnections = new Integer(5);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "Integer"; it is deprecated. ([squid:CallToDeprecatedMethod](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=138&issues=AWeegEgKh8YERCev6R1i&open=AWeegEgKh8YERCev6R1i)

[ISSUE_KEY]: AWeegEgKh8YERCev6R1i